tests/index.test.ts

```diff
@@ -436,8 +436,9 @@ interface MinProvider {
 function constructProviderOnlyContractCaller(
   provider: MinProvider,
   account?: string
-): ContractCallerFunctions<string> {
-  const staticCall: StaticContractCallerFn = async ({
+): Pick<ContractCallerFunctions<string>, 'transactCall'> {
+  // staticCall isn't currently necessary, because provider is only used in approveToken currently for tx making
```

Comment:
what's the idea behind static call at all? Do we need it, as in do we plan to do anything with it?

Reply:
Maaaybe. It can be useful later. If `contractCaller` is constructed fully manually it's optional (`approveToken` types require only `transactCall`).
Let's see if we use it later